src/ucp/api/ucp.h

```diff
@@ -1079,6 +1097,39 @@ struct ucp_tag_recv_info {
 };
 
 
+typedef struct {
+    /**
+     * Mask of valid fields in this structure and operation flags, using
+     * bits from @ref ucp_op_attr. Fields not specified in this mask will be
+     * ignored. Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t attr_mask;
+    /**
+     * Request handle allocated by the user. There should
+     * be at least UCP request size bytes of available
+     * space before the @a req. The size of UCP request
+     * can be obtained by @ref ucp_context_query function.
+     */
+    void    *request;
+    /**
+     * Callback function that is invoked whenever the
+     * send operation is completed.
+     */
+    union {
+        ucp_send_nbx_callback_t     send;
```

Comment:
align there type names?

Reply:
:+1: for `ucp_send_nbx_callback_t` name

Reply:
If `ucp_send_nbx_callback_t` and `ucp_tag_recv_nbx_callback_t` have the same signature, can we not have  `ucp_generic_nbx_callback_t` instead? Or is the purpose to be explicit about the request being used for a send/recv operation?

Reply:
as I can see there are different signatures (different incoming arguments), that is why different types are used

Reply:
Sorry. I've misread the changes. Please ignore my comment.

Reply:
send/recv - it will be better to rename to something generic that is not limited to send/recv. It used for other non-send/recv communication semantics.